File: elem/leap_year.js


const leap = () => {
    let date = new Date();
    let year = date.getFullYear();
    let years = [];

    for (; years.length < 20; year++) {
        if (year % 400 == 0 || year % 4 == 0 && year % 100 != 0) {
            years.push(year);
        }
    }

    for (y in years) {
        console.log(years[y]);
    }

}


leap();

File: elem/leap_year.py
import datetime 

def leap():
    x = datetime.datetime.now()
    yr = x.year 
    years = []

    while len(years) < 20:
        if yr % 400 == 0 or yr % 4 == 0 and yr % 100 != 0:
            years.append(yr)
        yr += 1

    for y in years:
        print(y)


leap()